### DISCUSSION TITLE
Can this work with Jacoco in maven on bitbucket?

### DISCUSSION THREAD

**Discussion by preritsarvaiya** [danielpalme/ReportGenerator#611] [Q&A] [answered]
Hello! I hope you're doing well. I have a few questions:
1. Can this be used with Jacoco in Java?
2. Is there a plugin available for Bitbucket that can display a report similar to this image: 
![image](https://github.com/danielpalme/ReportGenerator/assets/105106656/dfca5697-5d2a-4756-947f-45eca758aef6) ?
3. Could you please share the relevant Java documentation that makes the setup process easier for everyone?

**Comment by danielpalme** [comment]
1. Yes that's possible! See here: https://github.com/danielpalme/ReportGenerator/wiki/Getting-started#java
2. I'm not sure. There is a plugin, but I'm not the maintainer. So I don't know which features the plugin provides. See https://github.com/LoremFooBar/reportgenerator-coverage-bitbucket-pipe
3. See https://github.com/danielpalme/ReportGenerator/wiki/Getting-started#java

**Reply by preritsarvaiya** [reply]
Thank you for the response.

I attempted to use the solution provided at https://github.com/danielpalme/ReportGenerator/wiki/Getting-started#java. However, I encountered continuous failures in the pipeline when using it on Bitbucket.

**Reply by danielpalme** [reply]
What error message do get on Bitbucket?

**Reply by preritsarvaiya** [reply]
`pipelines:
  default:
    - step:
        name: Build and Test
        image: maven:3.8.4-openjdk-17-slim
        script:
          - mvn clean install
 
    - step:
        name: Generate Coverage Report
        image: mcr.microsoft.com/dotnet/sdk:5.0
        script:
          - mvn test jacoco:report
          - apt-get update && apt-get install -y wget unzip
          - wget -q https://github.com/danielpalme/ReportGenerator/releases/download/v5.1.23/ReportGenerator_5.1.23.zip
          - unzip -q ReportGenerator_5.1.23.zip -d tools
          - dotnet tools/reportgenerator -reports:target/jacoco-ut/jacoco.xml -targetdir:coveragereport -sourcedirs:src/main/java

    - step:
        name: Upload Coverage Report
        image: atlassian/pipelines-awscli
        script:
          - aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID
          - aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY
          - aws s3 cp coveragereport s3://mytest-reports/coveragereport --recursive

definitions:
  caches:
    maven: ~/.m2/repository`
    
    So this was the pipeline I wrote, it didn't work and it also didn't show the report in the bitbucket.

**Reply by danielpalme** [reply]
But the pipline does not fail?

**Reply by preritsarvaiya** [reply]
It failed and not working correctly

**Reply by danielpalme** [reply]
Which step fails? What error message is displayed?

**Reply by preritsarvaiya** [reply]
When Generating Coverage Report it is failing with this error
mvn test jacoco:report
bash: mvn: command not found

**Reply by danielpalme** [reply, marked as answer]
That indicates that Maven `mvn` is not installed. You can install it with this command:
`sudo apt install maven`

The Java SDK also has to be available.
